test/functional/api/cas/statistics.py

```diff
@@ -1,256 +1,218 @@
-            self.block_stats = BlockStats(
-                *[stats[stat] for stat in block_stats_core]
-            )
-        except KeyError:
-            pass
-        try:
-            self.error_stats = ErrorStats(
-                *[stats[stat] for stat in error_stats]
-            )
-        except KeyError:
-            pass
+    def __init__(
+        self,
+        cache_id: int,
+        core_id: int,
+        filter: List[StatsFilter] = None,
+        percentage_val: bool = False,
+    ):
+
+        if filter is None:
+            filters = [
+                StatsFilter.conf,
+                StatsFilter.usage,
+                StatsFilter.req,
+                StatsFilter.blk,
+                StatsFilter.err,
+            ]
+        else:
+            filters = filter
+
+        csv_stats = casadm.print_statistics(
+            cache_id=cache_id,
+            core_id=core_id,
+            filter=filter,
+            output_format=casadm.OutputFormat.csv,
+        ).stdout.splitlines()
+
+        stat_keys, stat_values = csv.reader(csv_stats)
+
+        # Unify names in block stats for core and cache:
+        # cache stats: Reads from core(s)
+        # core stats: Reads from core
+        stat_keys = [x.replace("(s)", "") for x in stat_keys]
+        stats_dict = dict(zip(stat_keys, stat_values))
+
+        for filter in filters:
+            match filter:
+                case StatsFilter.conf:
+                    self.config_stats = CoreConfigStats(stats_dict)
+                case StatsFilter.usage:
+                    self.usage_stats = UsageStats(stats_dict, percentage_val)
+                case StatsFilter.req:
+                    self.request_stats = RequestStats(stats_dict, percentage_val)
+                case StatsFilter.blk:
+                    self.block_stats_cache = BlockStats(stats_dict, percentage_val)
+                case StatsFilter.err:
+                    self.error_stats = ErrorStats(stats_dict, percentage_val)
 
     def __str__(self):
-        status = ""
-        for stats_item in self.stats_list:
-            current_stat = getattr(self, stats_item, None)
-            if current_stat:
-                status += f"--- Core {current_stat}"
-        return status
+        # stats_list contains all Class.__str__ methods initialized in CacheStats
+        stats_list = [str(getattr(self, stats_item)) for stats_item in self.__dict__]
+        return "\n".join(stats_list)
 
     def __eq__(self, other):
-        if not other:
-            return False
-        for stats_item in self.stats_list:
-            if getattr(self, stats_item, None) != getattr(other, stats_item, None):
-                return False
-        return True
+        # check if all initialized variable in self(CacheStats) match other(CacheStats)
+        return [getattr(self, stats_item) for stats_item in self.__dict__] == [
+            getattr(other, stats_item) for stats_item in other.__dict__
+        ]
 
 
 class IoClassStats:
```

Comment:
I like that you unified how IoClassStats work inside for Cache and Core, but...
I suggest renaming IoClassStats to CoreIoClassStats and add CacheIoClassStats(CoreIoClassStats) with
 \_\_init\_\_(<args without core_id>): 
super().__init__(<args with added core_id=None>)
I hope it's clear. What I hope to achieve here is a distinction between CacheIoClassStats and CoreIoClassStats.